src/operator/output.rs

```diff
@@ -0,0 +1,311 @@
+    /// clock cycle.
+    ///
+    /// This is equivalent to calling
+    /// [`take_from_worker`](`Self::take_from_worker`) for each
+    /// worker thread in order and storing all none-`None`
+    /// results in a vector.
+    pub fn take_from_all(&self) -> Vec<T> {
+        let num_workers = self.0.mailbox.len();
+        let mut res = Vec::with_capacity(num_workers);
+
+        for worker in 0..num_workers {
+            if let Some(v) = self.take_from_worker(worker) {
+                res.push(v);
+            }
+        }
+        res
+    }
+}
+
+impl<T> OutputHandle<T>
+where
+    T: Batch<Time = ()> + Send,
+{
+    /// Read batches produced by all worker threads during the last
+    /// clock cycle and consolidate them into a single batch.
+    ///
+    /// This method is used in the common case when the `OutputHandle` is
+    /// attached to a stream that carries [`Batch`](`crate::trace::Batch`)es
+    /// of updates to relational data.  Semantically, each `Batch` consists
+    /// of `(key, value, weight)` tuples.  Depending on the structure of the
+    /// circuit, the same `key` or `(key, value)` pair can occur in batches
+    /// produced by multiple workers.  This method retrieves batches
+    /// produced by all workers and consolidates them into a single batch
+    /// where each `(key, value)` pair occurs exactly once.
+    ///
+    /// Internally, `consolidate` calls `take_from_worker` to retrieve batches
+    /// from individual worker threads.  See
+    /// [`take_from_worker`](`Self::take_from_worker`) documentation for the
+    /// exact semantics of this method.  In particular, note that repeated calls
+    /// to `take_from_worker` return `None`. `consolidate` skips `None` results
+    /// when computing the consolidated batch.
+    pub fn consolidate(&self) -> T {
+        let batches = self.take_from_all();
+        let mut spine = Spine::new(None);
+
+        for batch in batches {
+            spine.insert(batch);
+        }
+
+        spine.consolidate().unwrap_or_else(|| T::empty(()))
+    }
+}
+
+/// Sink operator that stores the contents of its input stream in
+/// the `OutputHandle`.
+struct Output<T> {
+    mailbox: Mailbox<Option<T>>,
+}
+
+impl<T> Output<T>
+where
+    T: Clone + Send + 'static,
+{
+    fn new() -> (Self, OutputHandle<T>) {
+        let handle = OutputHandle::new();
+        let mailbox = handle.mailbox(Runtime::worker_index()).clone();
+
+        let output = Self { mailbox };
+
+        (output, handle)
+    }
+}
+
+impl<T> Operator for Output<T>
+where
+    T: 'static,
+{
+    fn name(&self) -> Cow<'static, str> {
+        Cow::from("Output")
+    }
+
+    fn fixedpoint(&self, _scope: Scope) -> bool {
+        true
+    }
+}
+
+impl<T> SinkOperator<T> for Output<T>
+where
+    T: Clone + 'static,
+{
+    fn eval(&mut self, val: &T) {
+        self.mailbox.set(Some(val.clone()));
+    }
+
+    fn eval_owned(&mut self, val: T) {
+        self.mailbox.set(Some(val));
+    }
```

Comment:
Doesn't this mean that any consumer can miss values? If I forget to receive at any given clock cycle and the circuit progresses to another clock cycle then I miss the previous value

Reply:
Correct, it's a synchronous circuit, so there's a well-defined time when the output is available.  If you choose not to read it, it gets overwritten.